src/sqlfluff/core/linter/linter.py

```diff
@@ -225,7 +226,11 @@ def _parse_tokens(
         return parsed, violations
 
     @staticmethod
-    def parse_noqa(comment: str, line_no: int):
+    def parse_noqa(
+        comment: str,
+        line_no: int,
+        rule_set: List[BaseRule],
```

Comment:
Nitpick: For this and the other modified functions, should we consider passing in `List[str]` rather than `List[BaseRule]`, since that's needed? This is mostly a pedantic concern about excessive coupling, as they call it.

From [Wikipedia](https://en.wikipedia.org/wiki/Coupling_(computer_programming)):
>Stamp coupling (data-structured coupling)
>Stamp coupling occurs when modules share a composite data structure and use only parts of it, possibly different parts (e.g., passing a whole record to a function that needs only one field of it).
In this situation, a modification in a field that a module does not need may lead to changing the way the module reads the record.

Reply:
Think I just had it there for convenience when I was adding the additional logic to this function but definitely makes sense to clean up the coupling as you suggest 👍